### ISSUE TITLE
Thinking about Data Quality

### ISSUE THREAD

**Issue by Jegelewicz** [ArctosDB/newsletter#9]
How much do you think about the quality of the data you put into Arctos? Maybe we should all think about it more...

[Quality paralysis: a biodiversity data disease](http://iphylo.blogspot.com/2018/09/guest-post-quality-paralysis.html)

I had quite a few discussions with Bob at SPNHC/TDWG. Arctos members are doing more than most to create community-wide data quality checks, but we probably need to do more. As I have pointed out to Bob, we also need more community resources so that we aren't all using different ways to say the same thing or different fields to store the same data. And of course, we all need a Data Curator...

**Comment by dustymc** [comment]
Arctos was built in no small part because other "systems" suffer from those things. 

*   duplicate records

We have these too, but Arctos provides a mechanism to clearly mark them as such. If they're not necessary, Arctos provides a redirect mechanism. AFAIK Arctos is unique in assigning globally unique resolvable identifiers, and without those this seems like an unsolvable problem. 

*    records with data items in the wrong fields

Arctos generally does not store data in "fields," but rather as relational data objects. This problem isn't possible in Arctos with data which can be standardized. Data which can't be fully standardized is still type-controlled - you can enter something weird in "tail length" but you'll have to do it as a number and controlled units. The contents of free-text fields are always up to the user (one of the many reasons I always look for alternatives when eg, someone proposes a new free-text attribute).

*  records with data items inappropriate for a given field (includes Chapman's "domain schizophrenia")

As above - this isn't possible when the "field" is eg, {agent_id (FK AGENT)}.

*  records with truncated data items

There would be no reason for this in Arctos. We do have character count limits but they're almost always bigger than incoming data, overfilling always results in an error rather than truncation, and our mechanism to attach Media provides a place for data of any size or shape.

 * records with items in one field disagreeing with items in another

This exists in Arctos, but we have checks to prevent common things and more obscure things are easy to find. (Here's 176 "female" + something about "teste" in reproductive data records: http://arctos.database.museum/SpecimenResults.cfm?reproductive_data=teste&sex=female)

*  character encoding errors and mojibake

It happens, but Arctos is UTF16 encoded - this could only come from external tools.

*   wildly erroneous dates and spatial coordinates

It happens, but there are checks (eg, began_date can't occur after ended_date), our coordinate generally come from spatial sources, I think most problems get noticed quickly in the interfaces (eg, edit locality suggests georeferences and provides a distance between asserted and suggested), and there are many discovery tools (map border color on specimendetail, the "chronological extent" search field, the 'find specimens not in the shape' option from geography, etc.).

 *  internally inconsistent formatting of dates, names and other data items (e.g. 48 variations on "sea level" in a single set of records)

This is one of the reasons we normalize. 

The things Chapman mentions in the comments are generally not possible in Arctos, but I'm less inclined to place the responsibility at the feet of a developer (who may be working with limited tools or simply doing what they've been paid to do) or a Curator (who probably didn't sign on to design databases). There is a CMS which avoids those problems available to Administrators, and it comes with a built-in community willing to serve as "data curators" for each other.

**Comment by Jegelewicz** [comment]
I agree we are the best at this and none of the issues will ever completely go away. One thing I know the collections I can manage have trouble with is

>

>  
>     character encoding errors and mojibake
> 
> It happens, but Arctos is UTF16 encoded - this could only come from external tools.

All of the tildes and accents come out as weird triangles or question marks.

**Comment by dustymc** [comment]
>weird triangles or question marks

Example?

**Comment by campmlc** [comment]
I just fixed one last week. "Puente Nununya" had triangles for each "n"
because the locality was uploaded with an enye.

**Comment by dustymc** [comment]
http://arctos.database.museum/SpecimenResults.cfm?spec_locality=%C3%B1 is 4739 specimens that have an ñ in locality. 

They look like this in Arctos

<img width="135" alt="screen shot 2018-09-15 at 6 43 47 pm" src="https://user-images.githubusercontent.com/5720791/45592025-59fded00-b917-11e8-95d3-5e811c02a8b0.png">

and downloads

<img width="141" alt="screen shot 2018-09-15 at 6 44 33 pm" src="https://user-images.githubusercontent.com/5720791/45592031-74d06180-b917-11e8-9b16-7fc82ab2181f.png">

Anything else is an indication of conversion errors - they were not properly converted to UTF before they were uploaded, or your UI doesn't properly support UTF (or both!).

**Comment by campmlc** [comment]
I assume most of these must be legacy bulkloads? Any way to search and
repair? I imagine if you do a search on Latin American localities there
would be quite a few.
